connecting_fronts.go

```diff
@@ -0,0 +1,54 @@
+package fronted
+
+import (
+	"context"
+)
+
+type connectingFronts interface {
+	onConnected(m Front)
+	connectingFront(context.Context) (Front, error)
+	size() int
+}
+
+type connecting struct {
+	// Create a channel of fronts that are connecting.
+	frontsCh chan Front
+}
+
+// Make sure that connectingFronts is a connectListener
+var _ connectingFronts = &connecting{}
+
+// newConnectingFronts creates a new ConnectingFronts struct with a channel of fronts that have
+// successfully connected.
+func newConnectingFronts(size int) *connecting {
+	return &connecting{
+		// We overallocate the channel to avoid blocking.
+		frontsCh: make(chan Front, size),
+	}
+}
```

Comment:
Right -- it would be empty, but with a large capacity based on this call from fronted.go:

`connectingFronts:    newConnectingFronts(4000),`

The idea is to just to make sure that it won't block on adding connecting fronts.